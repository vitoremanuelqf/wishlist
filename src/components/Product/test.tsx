import { render, screen } from '@testing-library/react'

import { Product } from '.'

jest.mock('../../hooks/useWishlist', () => {
  return {
    useWishlist() {
      return [
        {
          id: '1',
          alt: 'alt',
          src: 'src',
          title: 'Kit Meia Invisível Olympikus C/ 3 Pares Masculina',
          rating: 4,
          fullPrice: 'R$ 100,00',
          finalPrice: 'R$ 50,00',
        },
      ]
    },
  }
})

const productItem = {
  id: '1',
  alt: 'alt',
  src: 'src',
  title: 'Kit Meia Invisível',
  rating: 4,
  fullPrice: 'R$ 100,00',
  finalPrice: 'R$ 50,00',
}

describe('<Product />', () => {
  it('should render the product card on the home page', () => {
    const { container } = render(
      <Product data={productItem} productType={'home'} />,
    )

    expect(screen.getByAltText(/alt/i)).toBeInTheDocument()
    expect(screen.getByText(/Kit Meia Invisível/i)).toBeInTheDocument()
    expect(screen.getByText(/4\.0/i)).toBeInTheDocument()
    expect(screen.getByText(/R\$ 100,00/i)).toBeInTheDocument()
    expect(screen.getByText(/R\$ 50,00/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the product card in the wishlist', () => {
    const { container } = render(
      <Product data={productItem} productType={'wishlist'} />,
    )

    expect(screen.getByAltText(/alt/i)).toBeInTheDocument()
    expect(screen.getByText(/Kit Meia Invisível/i)).toBeInTheDocument()
    expect(screen.getByText(/4\.0/i)).toBeInTheDocument()
    expect(screen.getByText(/R\$ 100,00/i)).toBeInTheDocument()
    expect(screen.getByText(/R\$ 50,00/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
