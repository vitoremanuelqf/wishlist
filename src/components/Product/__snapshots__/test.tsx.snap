// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> should render the product card in the wishlist 1`] = `
<div
  class="relative flex h-auto w-full flex-col gap-2 rounded-md border border-gray-300 p-2 shadow-md"
>
  <div
    class="relative flex aspect-square h-auto w-full items-center justify-center rounded-md border border-gray-300"
  >
    <img
      alt="alt"
      class="absolute bottom-0 left-0 right-0 top-0 aspect-square w-full rounded-md"
      src="src"
    />
  </div>
  <h2
    class="line-clamp-2 block w-full text-ellipsis text-sm font-normal text-gray-800 sm:text-base"
  >
    Kit Meia Invisível
  </h2>
  <div
    class="mt-auto flex w-full items-center"
  >
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="text-gray-200"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      />
    </svg>
    <span
      class="ml-1 font-medium text-gray-600"
    >
      4
      .0
    </span>
  </div>
  <span
    class="text-sm font-normal text-gray-300 line-through sm:text-base"
  >
    R$ 100,00
  </span>
  <p
    class="text-base font-bold text-[#5b2b84] sm:text-lg"
  >
    R$ 50,00
  </p>
</div>
`;

exports[`<Product /> should render the product card on the home page 1`] = `
<div
  class="relative flex h-auto w-full flex-col gap-2 rounded-md border border-gray-300 p-2 shadow-md"
>
  <button
    class="bg-gray-400 absolute right-3 top-3 z-10 flex h-8 w-8 items-center justify-center rounded-full border shadow-sm transition-all duration-150 hover:h-9 hover:w-9 sm:h-9 sm:w-9 sm:hover:h-10 sm:hover:w-10"
  >
    <svg
      class="h-5 w-5 text-white sm:h-6 sm:w-6"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
      />
    </svg>
  </button>
  <div
    class="relative flex aspect-square h-auto w-full items-center justify-center rounded-md border border-gray-300"
  >
    <img
      alt="alt"
      class="absolute bottom-0 left-0 right-0 top-0 aspect-square w-full rounded-md"
      src="src"
    />
  </div>
  <h2
    class="line-clamp-2 block w-full text-ellipsis text-sm font-normal text-gray-800 sm:text-base"
  >
    Kit Meia Invisível
  </h2>
  <div
    class="mt-auto flex w-full items-center"
  >
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="fill-yellow-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
      />
    </svg>
    <svg
      class="text-gray-200"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
      />
    </svg>
    <span
      class="ml-1 font-medium text-gray-600"
    >
      4
      .0
    </span>
  </div>
  <span
    class="text-sm font-normal text-gray-300 line-through sm:text-base"
  >
    R$ 100,00
  </span>
  <p
    class="text-base font-bold text-[#5b2b84] sm:text-lg"
  >
    R$ 50,00
  </p>
</div>
`;
